docs: add roundtrip to glossary (#400)
unicode-org/message-format-wg: docs/glossary-and-resources.md

* **locale fallback** - offering a reasonable substitute locale when the requested locale's resources are not available. Results may vary depending on context (ex: audio vs. text vs. video).
* **locale matching** - computing the locale fallback.
* **resource** - files bundled with an application that are loaded in by the executable code.  UI strings, etc. and their locale-specific translations are typically stored as resources.

> * **roundtrip (RTT)** - the process of translating text into another language, then translating the result back into the original language.

Reviewer: This seems like an excellent way to grow our glossary, esp. as it seems that the meaning of this term is not immediately obvious. Here's my understanding:

```suggestion
* **roundtrip (RTT)** - the process of transforming a message into another format or represenation, then transforming it back into the original format.
```

Author: i more or less ripped off https://en.wikipedia.org/wiki/Round-trip_translation

shucks, is their definition too narrow? or is ours too wide? i sincerely do not know 😄

Reviewer: I think it's more that we're working in a rather specialized context. The original and most common use of this term in our context is with respect to messages in ICU MessageFormat (aka MF1) syntax, and their representation in an MF2 data model.

Because MF1 can include inline selectors and MF2 can't, an MF1-to-MF2 transformer needs to duplicate any leading and trailing content around an MF1 selector to each of its variants. If that message is then transformed back into MF1, information is lost about its exact original syntax, but the resulting MF1 message will still format exactly the same as before.

Early on, an [explicit choice](https://github.com/unicode-org/message-format-wg/blob/main/docs/consensus_decisions.md#4-only-allow-for-selectors-at-the-top-level-of-a-message) was made to be ok with this.

Author: if you think others would be fine _overloading_ the definition just for this project, fine by me :) i generally would try to avoid overloading the term if its meaning is well known to the target audience... but perhaps it's close enough to not make a fuss!
